zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterOutput.java

```diff
@@ -229,8 +229,11 @@ public void write(int b) throws IOException {
               return;
             }
           }
-          // not a defined display system
+          // not a defined display system, probably a spell type
+          setType(InterpreterResult.Type.SPELL);
```

Comment:
IMO, we can pass available spell magics (e.g `%markdown`, `%sigma`) whenever executing paragraph. And it can be passed into `InterpreterOutput`. Then, we don't need to worry about.